Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -80,7 +80,31 @@ public bool isServer
         /// <para>This value is determined at runtime, as opposed to localPlayerAuthority which is set on the prefab. For most objects, authority is held by the server / host. For objects with localPlayerAuthority set, authority is held by the client of that player.</para>
         /// <para>For objects that had their authority set by AssignClientAuthority on the server, this will be true on the client that owns the object. NOT on other clients.</para>
         /// </summary>
-        public bool hasAuthority { get; private set; }
+        private bool isOwner;
+
+        public bool hasAuthority
+        {
+            get => isOwner;
+            set
+            {
+                bool prev = isOwner;
+                isOwner = value;
+
+                if (prev && !isOwner)
+                {
+                    OnStopAuthority();
+                }
+                if (!prev && isOwner)
+                {
+                    OnStartAuthority();
+                }
+            }
+        }
+
+        // whether this object has been spawned with authority
+        // we need hasAuthority and pendingOwner because
+        // we need to wait until all of them spawn before updating hasAuthority
+        internal bool pendingOwner { get; set; }
```

Comment:
does 'pendingOwner' make most sense? are we really waiting for owner, or are we waiting for authority? in which case 'pendingAuthority' might be more obvious